reframe/core/schedulers/slurm.py

```diff
@@ -318,6 +326,10 @@ def _update_state(self, job):
         if slurm_state_completed(job.state):
             # Since Slurm exitcodes are positive take the maximum one
             job.exitcode = max(int(s.group('exitcode')) for s in state_match)
+            self._completion_time = max(
+                datetime.strptime(s.group('end'), '%Y-%m-%dT%H:%M:%S')
```

Comment:
Are you sure that a parsable `end` time is always there? Even if the job is still running or pending?

Reply:
To make things more Pythonic and compact I'd also rewrite this as follows:

```python
parse_time = functools.partial(datetime.strptime, format='...')
self._completion_time = max(parse_time(s.group('end')) for s in state_match)
```

Reply:
> Are you sure that a parsable `end` time is always there?

I have tried it on daint and while it is running or pending the `end` time is `Unknown`. But the matched string is only parsed after the job is completed.

Reply:
So `strptime()` does not fail if you pass it a string that does not represent time then, right? Then it's fine.

Reply:
It does fail. But it is inside this if `if slurm_state_completed(job.state):`. It shouldn't be parsed while it is pending or running.

Reply:
Should I put this on a try? Because I don't know if I can test every state that is considered as completed.

Reply:
Ok, now I see it. That's fine.

Reply:
> Should I put this inside a try? Because I don't know if I can test every state that is considered as completed.

No, I haven't seen you were actually calling this inside the `if slurm_state_completed(job.state)`.

Reply:
> To make things more Pythonic and compact I'd also rewrite this as follows:
> 
> ```python
> parse_time = functools.partial(datetime.strptime, format='...')
> self._completion_time = max(parse_time(s.group('end')) for s in state_match)
> ```

Strptime doesn't have keyword arguments and I think I cannot set the second argument of the function.

Reply:
I think it's because it uses the C library behind the scenes. That's fine.